### ISSUE TITLE
Dynamic component loading from a pathname in shallow routing

### ISSUE THREAD

**Issue by mquandalle** [sveltejs/kit#13124]
### Describe the problem

When implementing shallow routing with modals in SvelteKit, we currently need to statically import page components:

https://svelte.dev/docs/kit/shallow-routing#Loading-data-for-a-route

```svelte
<script>
  import { preloadData } from '$app/navigation';
  import Modal from "$lib/Modal.svelte"
  import PhotoPage from './[id]/+page.svelte'; // Static import required, could also be `await import(...)`

  let modalOpened = $state(false);
  let modalData = $state();

  function modalNavigation() {
    modalData = await preloadData(href); // I can get the data from href, but not the related `+page.svelte` component
  }
</script>

{#if modalOpened}
  <Modal>
    <PhotoPage data={modalData} />
  </Modal>
{/if}
```

While `preloadData()` allows us to dynamically fetch route data, there's no equivalent for loading the page component itself. This becomes problematic when handling multiple shallow routes, as each `+page.svelte` component must be imported statically at build time.

### Describe the proposed solution

I'd like to dynamically resolve page components at runtime, similar to how we fetch data:

```ts
const pathname = "/photo/abc";
const modalData = await preloadData(pathname);
const ModalComponent = await preloadPage(pathname);
```

This would simplify shallow routing implementations, especially when dealing with multiple potential routes that could be opened in a modal.

### Alternatives considered

The signature of `preloadCode` could be changed to return the `+page.svelte` component

### Importance

nice to have

**Comment by david-plugge** [comment]
You can use a dynamic import:

```svelte
<script>
  async function load() {
    const data = await preloadData(href);
    const component = await import('./[id]/+page.svelte');
    return {
      data,
      component
    }
  }
</script>
```

But i do wonder how your page component looks like if you can reuse it within the modal. I typically use a specific component for such use cases that i use on the modal and the actual page.

**Comment by mquandalle** [comment]
Yes I know that I can use a dynamic import but I still need to reference the `./[id]/+page.svelte` path in my source code. In my case I have many links to different routes that I open in a modal, so instead I would like the SvelteKit router to resolve these components from a `href`.

**Comment by tv42** [comment]
A second use case that sounds similar: I was pondering letting the user manage multiple panes within one browser window, where each pane can independently navigate to different components. It would be nice if I could use a "fake URL" for each pane (say, store all of them in query args) and call `const component = lookupRoute(path)` for each and just render those components in panes, letting them take params from the path and run their own loaders etc. Right now it feels like I'd be reimplementing routing to avoid hardcoding a list of known components in my pane manager.
